src/sqlfluff/utils/reflow/elements.py

```diff
@@ -0,0 +1,383 @@
+                            LintFix(
+                                "replace",
+                                anchor=ws_seg,
+                                edit=[new_seg],
+                            )
+                        )
+                        segment_buffer[ws_idx] = new_seg
+
+                else:
+                    raise NotImplementedError(  # pragma: no cover
+                        f"Unexpected Constraints: {pre_constraint}, {post_constraint}"
+                    )
+            else:
+                # No. Should we insert some?
+
+                # Do we have either side set to "touch" or "any"
+                if {"touch", "any"}.intersection([pre_constraint, post_constraint]):
+                    # In this instance - no whitespace is correct.
+                    # Either because there shouldn't be, or because "any"
+                    # means we shouldn't check.
+                    pass
+                # Handle the default case
+                elif pre_constraint == post_constraint == "single":
+                    # Insert a single whitespace.
+                    reflow_logger.debug("    Inserting Single Whitespace.")
+                    # Add it to the buffer first (the easy bit)
+                    segment_buffer = [WhitespaceSegment()]
+
+                    # So special handling here. If segments either side
+                    # already exist then we don't care which we anchor on
+                    # but if one is already an insertion (as shown by a lack)
+                    # of pos_marker, then we should piggy back on that pre-existing
+                    # fix.
+                    existing_fix = None
+                    insertion = None
+                    if prev_block and not prev_block.segments[-1].pos_marker:
+                        existing_fix = "after"
+                        insertion = prev_block.segments[-1]
+                    elif next_block and not next_block.segments[0].pos_marker:
+                        existing_fix = "before"
+                        insertion = next_block.segments[0]
+
+                    if existing_fix:
+                        reflow_logger.debug(
+                            "    Detected existing fix %s", existing_fix
+                        )
+                        if not fixes:  # pragma: no cover
+                            raise ValueError(
+                                "Fixes detected, but none passed to .respace(). "
+                                "This will cause conflicts."
+                            )
+                        # Find the fix
+                        for fix in fixes:
+                            # Does it contain the insertion?
+                            # TODO: This feels ugly - eq for BaseSegment is different
+                            # to uuid matching for RawSegment. Perhaps this should be
+                            # more aligned. There might be a better way of doing this.
```

Comment:
Would it be safe to use `is` here (checking for same exact object)?

Reply:
Unfortunately not - we're not checking for _exactly_ the same segment, because that includes the location which might change when fixes are applied. We're checking for a segment which has the same signature rather than the same object.